realm/src/main/java/io/realm/Realm.java

```diff
@@ -1206,7 +1206,12 @@ public void addChangeListener(RealmChangeListener listener) {
      */
     public void removeChangeListener(RealmChangeListener listener) {
         checkIfValid();
-        changeListeners.remove(new WeakReference<RealmChangeListener>(listener));
+        Iterator<WeakReference<RealmChangeListener>> iterator = changeListeners.iterator();
+        while (iterator.hasNext()) {
```

Comment:
As you pointed out in a comment (I can't find anymore?) The current implementation has a chance of ConcurrentModificationException. This can still happen using this code:

eg. the following test case crashes:

        final AtomicInteger counter= new AtomicInteger(0);
        RealmChangeListener listener = new RealmChangeListener() {
            @Override
            public void onChange() {
                counter.incrementAndGet();
                realm.addChangeListener(this);
            }
        };

        realm = Realm.getInstance(getContext());
        realm.addChangeListener(listener);

        realm.beginTransaction();
        realm.createObject(AllTypes.class);
        realm.commitTransaction();

        realm.beginTransaction();
        realm.createObject(AllTypes.class);
        realm.commitTransaction();

        assertEquals(2, counter.get());